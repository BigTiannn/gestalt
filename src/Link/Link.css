.link {
  composes: bold from global;
  text-decoration: none;
}

.link:hover,
.link:focus {
  text-decoration: underline;
}

.inline {
  /* dom default */
}

.block {
  composes: block from global;
}

/* colors */

.white {
  composes: white from global;
}

.darkGray {
  composes: dark-gray from global;
}

.gray {
  composes: gray from global;
}

.red {
  composes: red from global;
}

.blue {
  composes: blue from global;
}
