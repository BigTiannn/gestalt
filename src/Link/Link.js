// @flow
import React, { PropTypes } from 'react';
import cx from 'classnames';
import styles from './Link.css';

const contains = (key, arr) => arr.indexOf(key) >= 0;
const omit = (keys, obj) => Object.keys(obj).reduce((acc, k) => {
  if (contains(k, keys)) {
    return acc;
  }
  return {
    ...acc,
    [k]: obj[k],
  };
}, {});

const PropBlacklist = [
  'aria-hidden',
  'className',
  'color',
  'inline',
  'style',
];

type LinkProps = {|
  // eslint-disable-next-line react/no-unused-prop-types
  children?: any,
  color: 'white' | 'darkGray' | 'gray' | 'red' | 'blue',
  // eslint-disable-next-line react/no-unused-prop-types
  href: string,
  inline?: boolean,
|};

type GestaltContext = {
  inputDevice: '' | 'key' | 'mouse' | 'touch'
};

export default function Link(props: LinkProps, context: GestaltContext) {
  const { inputDevice = 'key' } = context;
  const { color = 'darkGray', inline = false } = props;
  return (
    // eslint-disable-next-line jsx-a11y/anchor-has-content
    <a
      {...omit(PropBlacklist, props)}
      className={
        cx(
          styles.link,
          styles[color],
          (inline ? styles.inline : styles.block),
          (inputDevice !== 'key' ? styles.disableFocusOutline : '')
        )
      }
    />
  );
}

Link.contextTypes = {
  inputDevice: React.PropTypes.string
};

Link.propTypes = {
  // eslint-disable-next-line react/no-unused-prop-types
  children: PropTypes.node.isRequired,
  color: PropTypes.oneOf(['white', 'darkGray', 'gray', 'red', 'blue']),
  // eslint-disable-next-line react/no-unused-prop-types
  href: PropTypes.string.isRequired,
  inline: PropTypes.bool
};
