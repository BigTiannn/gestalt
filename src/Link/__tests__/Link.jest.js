/* eslint-env jest */
/* eslint import/imports-first: 0 */
jest.unmock('../../Link/Link');

import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../Link';

const snapshot = (component) => {
  const tree = renderer.create(
    component
  ).toJSON();
  expect(tree).toMatchSnapshot();
};

it('default', () => snapshot(<Link href="https://example.com">Link</Link>));
it('gray', () =>
  snapshot(<Link href="https://example.com" color="gray">Link</Link>)
);
it('gray inline', () =>
  snapshot(<Link href="https://example.com" color="gray" inline>Link</Link>)
);
it('darkGray', () =>
  snapshot(<Link href="https://example.com" color="darkGray">Link</Link>)
);
it('darkGray inline', () =>
  snapshot(<Link href="https://example.com" color="darkGray" inline>Link</Link>)
);
it('white', () =>
  snapshot(<Link href="https://example.com" color="white">Link</Link>)
);
it('white inline', () =>
  snapshot(<Link href="https://example.com" color="white" inline>Link</Link>)
);
it('red', () =>
  snapshot(<Link href="https://example.com" color="red">Link</Link>)
);
it('red inline', () =>
  snapshot(<Link href="https://example.com" color="red" inline>Link</Link>)
);
it('blue', () =>
  snapshot(<Link href="https://example.com" color="blue">Link</Link>)
);
it('blue inline', () =>
  snapshot(<Link href="https://example.com" color="blue" inline>Link</Link>)
);
