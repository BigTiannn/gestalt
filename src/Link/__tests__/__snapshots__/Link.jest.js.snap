exports[`test blue 1`] = `
<a
  className="link blue block"
  href="https://example.com">
  Link
</a>
`;

exports[`test blue inline 1`] = `
<a
  className="link blue inline"
  href="https://example.com">
  Link
</a>
`;

exports[`test darkGray 1`] = `
<a
  className="link darkGray block"
  href="https://example.com">
  Link
</a>
`;

exports[`test darkGray inline 1`] = `
<a
  className="link darkGray inline"
  href="https://example.com">
  Link
</a>
`;

exports[`test default 1`] = `
<a
  className="link darkGray block"
  href="https://example.com">
  Link
</a>
`;

exports[`test gray 1`] = `
<a
  className="link gray block"
  href="https://example.com">
  Link
</a>
`;

exports[`test gray inline 1`] = `
<a
  className="link gray inline"
  href="https://example.com">
  Link
</a>
`;

exports[`test red 1`] = `
<a
  className="link red block"
  href="https://example.com">
  Link
</a>
`;

exports[`test red inline 1`] = `
<a
  className="link red inline"
  href="https://example.com">
  Link
</a>
`;

exports[`test white 1`] = `
<a
  className="link white block"
  href="https://example.com">
  Link
</a>
`;

exports[`test white inline 1`] = `
<a
  className="link white inline"
  href="https://example.com">
  Link
</a>
`;
